### PR TITLE
Made http urls and type names case insensitive

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: zeq/http/server.cpp [hunk 1]
```diff
@@ -225,6 +226,15 @@ class Server::Impl : public detail::Sender
...
             response.set_status( 400 );
     }
+
+private:
+    std::string _toLower( const std::string& value )
+    {
+        std::string str( value );
+        std::transform( str.begin(), str.end(), str.begin(), ::tolower );
+        return str;
+    }
```
Comment (eile) [inline zeq/http/server.cpp]: Can be a free function in an anonymous namespace.

File: zeq/http/server.cpp [hunk 2]
```diff
@@ -45,7 +46,7 @@ class Server::Impl : public detail::Sender
 
     bool subscribe( servus::Serializable& serializable )
     {
-        const std::string& name = serializable.getTypeName();
+        const std::string name = _toLower( serializable.getTypeName( ));
```
Comment (eile) [inline zeq/http/server.cpp]: & can stay

**Comment by bilgili** [comment]
Why do we need this change ? We are not following any standards but those are case insensitive in real world ?

**Comment by favreau** [comment]
@bilgili There are lots of discussions around and the answer depends on the OS hosting the service. It seems to me that the majority of APIs around the web are case insensitive, and I personally tend to agree with this policy (http://stackoverflow.com/questions/7996919/should-url-be-case-sensitive).

**Comment by bilgili** [comment]
fair enough

**Comment by eile** [comment]
Test is missing.

File: zeq/http/server.cpp [hunk 3]
```diff
@@ -14,9 +14,20 @@
...
 #include <httpxx/ResponseBuilder.hpp>
+#include <algorithm>
 
 namespace httpxx = ::http; // avoid confusion between httpxx and zeq::http
 
+namespace
+{
+std::string _toLower( const std::string& value )
+{
+    std::string str( value );
```
Comment (eile) [inline zeq/http/server.cpp]: rm this line, pass value by copy.

**Comment by eile** [comment]
+1 otherwise

**Comment by eile** [comment]
+1, rebase master missing.

**Pull request merged**